src/index.js

```diff
@@ -202,172 +286,266 @@ class PeerId {
+      throw new Error('Supplied key is neither a base64 string nor a Uint8Array')
+    }
 
-// Private key input will be a string
-exports.createFromPrivKey = async (key) => {
-  if (typeof key === 'string') {
-    key = uint8ArrayFromString(key, 'base64pad')
+    const privKey = await cryptoKeys.unmarshalPrivateKey(key)
+    return computePeerId(privKey.public, privKey)
   }
 
-  if (!(key instanceof Uint8Array)) {
-    throw new Error('Supplied key is neither a base64 string nor a Uint8Array')
-  }
+  /**
+   * Create PeerId from PeerId JSON formatted object.
+   * 
+   * @param {PeerIdJSON} obj 
+   * @returns {Promise<PeerId>}
+   */
+  static async createFromJSON (obj) {
+    const id = mh.fromB58String(obj.id)
+    const rawPrivKey = obj.privKey && uint8ArrayFromString(obj.privKey, 'base64pad')
+    const rawPubKey = obj.pubKey && uint8ArrayFromString(obj.pubKey, 'base64pad')
+    const pub = rawPubKey && await cryptoKeys.unmarshalPublicKey(rawPubKey)
+
+    if (!rawPrivKey) {
+      return new PeerId(id, undefined, pub)
+    }
 
-  const privKey = await cryptoKeys.unmarshalPrivateKey(key)
-  return computePeerId(privKey, privKey.public)
-}
+    const privKey = await cryptoKeys.unmarshalPrivateKey(rawPrivKey)
+    const privDigest = await computeDigest(privKey.public)
 
-exports.createFromJSON = async (obj) => {
-  const id = mh.fromB58String(obj.id)
-  const rawPrivKey = obj.privKey && uint8ArrayFromString(obj.privKey, 'base64pad')
-  const rawPubKey = obj.pubKey && uint8ArrayFromString(obj.pubKey, 'base64pad')
-  const pub = rawPubKey && await cryptoKeys.unmarshalPublicKey(rawPubKey)
+    /**
+     * @type {Uint8Array}
+     */
+    let pubDigest = new Uint8Array()
 
-  if (!rawPrivKey) {
-    return new PeerIdWithIs(id, null, pub)
-  }
+    if (pub) {
+      pubDigest = await computeDigest(pub)
+    }
 
-  const privKey = await cryptoKeys.unmarshalPrivateKey(rawPrivKey)
-  const privDigest = await computeDigest(privKey.public)
+    if (pub && !uint8ArrayEquals(privDigest, pubDigest)) {
+      throw new Error('Public and private key do not match')
+    }
 
-  let pubDigest
+    if (id && !uint8ArrayEquals(privDigest, id)) {
+      throw new Error('Id and private key do not match')
+    }
 
-  if (pub) {
-    pubDigest = await computeDigest(pub)
+    return new PeerId(id, privKey, pub)
   }
 
-  if (pub && !uint8ArrayEquals(privDigest, pubDigest)) {
-    throw new Error('Public and private key do not match')
-  }
+  /**
+   * Create PeerId from Protobuf bytes. 
+   * 
+   * @param {Uint8Array | string} buf - Protobuf bytes, as Uint8Array or hex-encoded string.
+   * @returns {Promise<PeerId>}
+   */
+  static async createFromProtobuf (buf) {
+    if (typeof buf === 'string') {
+      buf = uint8ArrayFromString(buf, 'base16')
+    }
 
-  if (id && !uint8ArrayEquals(privDigest, id)) {
-    throw new Error('Id and private key do not match')
-  }
+    let { id, privKey: privKeyArray, pubKey: pubKeyArray } = PeerIdProto.decode(buf)
```

Comment:
Probably better to call this privKeyBytes and pubKeyBytes. I was not undertanding why an array of private keys here, but then I understood that this was probably for being an uint8array.